File: Kotlin/Aula-Kotlin-24-08-22/src/main/kotlin/classesabstratas/Animal.kt
package classesabstratas

// abstract - ser aberta, porem não podera criar instancia direto da class Animal

abstract class Animal(
    val especie: String,
    val nomeCientifico: String,
    val habitat: String,
    val carnivoro: Boolean,
    val sexo: String,
    val idade: Int
        ){

    abstract fun correr()

    abstract fun dormir()

    abstract fun andar()

    abstract fun comer()

    abstract fun emitirSom()



}

File: JavaScript-TypeScript/Ex-JS-TS-06-09-22/src/animal.ts
export abstract class Animal{

    nome: string
    raca: string
    idade: number

    constructor(
        nome: string,
        raca: string,
        idade: number
    ){
        this.nome = nome
        this.raca = raca
        this.idade = idade
    }

    abstract seMovimentar(): void

    abstract comer(): void

    abstract transmitirSom(): void
}